doc: Use rel="canonical" attribute when linking to "latest" docs pages

kartben opened the pull request:
Help search engines identify "latest" version of the documentation as the canonical source.
Fixes #54749.

It would be great if this can make it into 3.3.

stephanosio commented on doc/_templates/layout.html:
```diff
@@ -2,7 +2,7 @@
   {% if is_release %}
     <div class="wy-alert wy-alert-danger">
-      The <a href="/latest/{{ pagename }}.html">latest development version</a>
+      The <a href="/latest/{{ pagename }}.html" rel="canonical">latest development version</a>
```
Is this really a correct way to specify `rel="canonical"`?

Various sources, including the [original RFC](https://www.ietf.org/rfc/rfc6596.txt), only mention the `link` tag (in `head`) as the place where this attribute should be specified.

kartben commented:
You are absolutely correct! My bad.
I will update the patch accordingly.

Le lun. 13 févr. 2023, 12:37, Stephanos Ioannidis ***@***.***>
a écrit :

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In doc/_templates/layout.html
> <https://github.com/zephyrproject-rtos/zephyr/pull/54767#discussion_r1104351291>
> :
>
> > -      The <a href="/latest/{{ pagename }}.html">latest development version</a>
> +      The <a href="/latest/{{ pagename }}.html" rel="canonical">latest development version</a>
>
> Is this really a correct way to specify rel="canonical"?
>
> Various sources, including the original RFC
> <https://www.ietf.org/rfc/rfc6596.txt>, only mention the link tag (in head)
> as the place where this attribute should be specified.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zephyrproject-rtos/zephyr/pull/54767#pullrequestreview-1295471304>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA7J64O2HDCBP6CM5OP7LTWXIMG7ANCNFSM6AAAAAAU2CPCMA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kartben committed:
doc: set documentation base URL to /latest

Help search engines flag "latest" version of the documentation as
the canonical source by setting Sphinx's html_baseurl.
Fixes #54749.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

stephanosio reviewed (approved):
[Seems to be working](https://builds.zephyrproject.io/zephyr/pr/54767/docs/index.html):

```html
<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Zephyr Project Documentation &mdash; Zephyr Project Documentation</title>
      <link rel="stylesheet" href="[_static/pygments.css](https://builds.zephyrproject.io/zephyr/pr/54767/docs/_static/pygments.css)" type="text/css" />
      <link rel="stylesheet" href="[_static/css/theme.css](https://builds.zephyrproject.io/zephyr/pr/54767/docs/_static/css/theme.css)" type="text/css" />
      <link rel="stylesheet" href="[_static/graphviz.css](https://builds.zephyrproject.io/zephyr/pr/54767/docs/_static/graphviz.css)" type="text/css" />
      <link rel="stylesheet" href="[_static/copybutton.css](https://builds.zephyrproject.io/zephyr/pr/54767/docs/_static/copybutton.css)" type="text/css" />
      <link rel="stylesheet" href="[_static/css/custom.css](https://builds.zephyrproject.io/zephyr/pr/54767/docs/_static/css/custom.css)" type="text/css" />
      <link rel="stylesheet" href="[_static/tabs.css](https://builds.zephyrproject.io/zephyr/pr/54767/docs/_static/tabs.css)" type="text/css" />
    <link rel="shortcut icon" href="[_static/favicon.png](https://builds.zephyrproject.io/zephyr/pr/54767/docs/_static/favicon.png)"/>
    <link rel="canonical" href="https://docs.zephyrproject.org/latest/index.html" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
```

stephanosio commented:
@gmarull @carlescufi Can you please take a look?

gmarull commented on doc/conf.py:
```diff
@@ -137,6 +137,7 @@
     "logo_only": True,
     "prev_next_buttons_location": None
 }
+html_baseurl = "https://docs.zephyrproject.org/latest/"
```
shouldn't this be changed to
```suggestion
html_baseurl = f"https://docs.zephyrproject.org/{version if is_release else 'latest'}/"
```

kartben replied:
Since released versions have the "disclaimer popup" saying "The latest development version of this page may be more current than this released x.y.z version." I would argue that latest should always be the canonical?

gmarull replied:
right
